src/WakuRlnV2.sol

```diff
@@ -81,177 +81,183 @@ contract WakuRlnV2 is Initializable, Ownable2StepUpgradeable, UUPSUpgradeable, M
-        if (memberExists(idCommitment)) revert DuplicateIdCommitment();
+    /// @notice Register a membership
+    /// @param idCommitment The idCommitment of the new membership
+    /// @param rateLimit The rate limit of the new membership
+    function register(uint256 idCommitment, uint32 rateLimit) external onlyValidIdCommitment(idCommitment) {
+        // FIXME: turn into modifier?
+        if (membershipExists(idCommitment)) revert DuplicateIdCommitment();
 
         uint32 index;
-        bool reusedIndex;
-        (index, reusedIndex) = _acquireMembership(_msgSender(), idCommitment, userMessageLimit, true);
+        bool indexReused;
+        (index, indexReused) = _acquireMembership(_msgSender(), idCommitment, rateLimit);
 
-        _register(idCommitment, userMessageLimit, index, reusedIndex);
+        _register(idCommitment, rateLimit, index, indexReused);
     }
 
-    /// @notice Allows a user to register as a member
-    /// @param idCommitment The idCommitment of the member
-    /// @param userMessageLimit The message limit of the member
-    /// @param membershipsToErase List of expired idCommitments to erase
+    /// @notice Register a membership
+    /// @param idCommitment The idCommitment of the new membership
+    /// @param rateLimit The rate limit of the new membership
+    /// @param idCommitmentsToErase List of idCommitments of expired memberships to erase
     function register(
         uint256 idCommitment,
-        uint32 userMessageLimit,
-        uint256[] calldata membershipsToErase
+        uint32 rateLimit,
+        uint256[] calldata idCommitmentsToErase
     )
         external
         onlyValidIdCommitment(idCommitment)
     {
-        if (memberExists(idCommitment)) revert DuplicateIdCommitment();
-
-        for (uint256 i = 0; i < membershipsToErase.length; i++) {
-            uint256 idCommitmentToErase = membershipsToErase[i];
-            MembershipInfo memory mdetails = members[idCommitmentToErase];
-            if (mdetails.userMessageLimit == 0) revert InvalidIdCommitment(idCommitmentToErase);
-            _eraseMembership(_msgSender(), idCommitmentToErase, mdetails);
-            LazyIMT.update(imtData, 0, mdetails.index);
+        // FIXME: turn into modifier?
+        if (membershipExists(idCommitment)) revert DuplicateIdCommitment();
+
+        // FIXME: extract in a separate function?
+        for (uint256 i = 0; i < idCommitmentsToErase.length; i++) {
+            uint256 idCommitmentToErase = idCommitmentsToErase[i];
+            MembershipInfo memory membershipToErase = memberships[idCommitmentToErase];
+            if (membershipToErase.rateLimit == 0) revert InvalidIdCommitment(idCommitmentToErase);
+            _eraseMembershipAndSaveSlotToReuse(_msgSender(), idCommitmentToErase, membershipToErase);
+            LazyIMT.update(merkleTree, 0, membershipToErase.index);
         }
 
+        // FIXME: code repeats cf. register()
```

Comment:
I would only have one `register` function, where you just pass `[]` when you don't want to specify ids to override, no?

Or, just handle `idCommentToErase` here and then call `register` without them once deleted.

Reply:
yeah, i guess it would make the contract usage from the app easier instead of having to choose between one function and the other...

Reply:
https://github.com/waku-org/waku-rlnv2-contract/pull/14/commits/b78a4c02cb13b4a597e7022e0359d85d1bfb32b5